editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/creators/AbstractClassBasedParameterPropertyUICreator.java

```diff
@@ -0,0 +1,188 @@
+				classToLoad = child.loadClass(parameter.getJavaType().substring(0, parameter.getJavaType().indexOf("<")));
+			} else {
+				classToLoad = child.loadClass(parameter.getJavaType());
+			}
+		} catch (ClassNotFoundException ex) {
+			CamelEditorUIActivator.pluginLog().logWarning("Cannot find class " + parameter.getJavaType() + " on classpath.", ex);
+			classToLoad = null;
+		}
+		return classToLoad;
+	}
+
+	/**
+	 * @param parent
+	 * @param project
+	 * @param fClass
+	 */
+	private void createBrowseButton(Composite parent, final IProject project, final Class<?> fClass) {
+		Button btn_browse = getWidgetFactory().createButton(parent, "...", SWT.FLAT | SWT.PUSH);
+		btn_browse.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				try {
+					IJavaProject javaProject = (IJavaProject) project.getNature(JavaCore.NATURE_ID);
+					IJavaElement[] elements = new IJavaElement[] { javaProject };
+					IJavaSearchScope scope = SearchEngine.createJavaSearchScope(elements);
+
+					FilteredTypesSelectionDialog dlg = new FilteredTypesSelectionDialog(Display.getDefault().getActiveShell(), false,
+							PlatformUI.getWorkbench().getProgressService(), scope, IJavaSearchConstants.CLASS);
+
+					if (Window.OK == dlg.open()) {
+						Object o = dlg.getFirstResult();
+						if (o instanceof SourceType) {
+							getControl().setText(((SourceType) o).getFullyQualifiedName());
+						} else if (o instanceof BinaryType) {
+							getControl().setText(((BinaryType) o).getFullyQualifiedName());
+						}
+					}
+				} catch (Exception ex) {
+					CamelEditorUIActivator.pluginLog().logError(ex);
+				}
+			}
+		});
+		btn_browse.setLayoutData(new GridData(SWT.RIGHT, SWT.CENTER, false, false, 1, 1));
+		btn_browse.setEnabled(fClass != null);
+	}
+
+	/**
+	 * @param parent
+	 * @param project
+	 * @param fClass
+	 */
+	private void createCreateButton(Composite parent, final IProject project, final Class<?> fClass) {
+		Button btn_create = getWidgetFactory().createButton(parent, " + ", SWT.FLAT | SWT.PUSH);
+		btn_create.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				NewClassCreationWizard wiz = new NewClassCreationWizard();
+				wiz.addPages();
+				wiz.init(PlatformUI.getWorkbench(), null);
+				NewClassWizardPage wp = (NewClassWizardPage) wiz.getStartingPage();
+				WizardDialog wd = new WizardDialog(e.display.getActiveShell(), wiz);
+				if (fClass.isInterface()) {
+					wp.setSuperInterfaces(Arrays.asList(fClass.getName()), true);
+				} else {
+					wp.setSuperClass(fClass.getName(), true);
+				}
+				wp.setAddComments(true, true);
+				IPackageFragmentRoot fragroot = null;
+				try {
+					IJavaProject javaProject = (IJavaProject) project.getNature(JavaCore.NATURE_ID);
+					IMavenProjectFacade facade = MavenPlugin.getMavenProjectRegistry().create(project, new NullProgressMonitor());

```

Comment:
We might want to change this to use the java classpath instead of the maven one. See https://github.com/fusesource/fuseide/commit/9040b132c0e306c1b462c9d8ea51a010fa41aa85